pkg/daemon/update.go

```diff
@@ -100,17 +104,58 @@ func (dn *Daemon) reconcilable(oldConfig, newConfig *mcfgv1.MachineConfig) (bool
 	if !reflect.DeepEqual(oldIgn.Networkd, newIgn.Networkd) {
 		glog.Warningf("daemon can't reconcile state!")
 		glog.Warningf("Ignition networkd section contains changes")
-		return false, nil
+		return false, nil, nil
 	}
 
 	// Passwd section
 
 	// we don't currently configure groups or users in place. we can't fix it if
 	// something changed here.
 	if !reflect.DeepEqual(oldIgn.Passwd, newIgn.Passwd) {
-		glog.Warningf("daemon can't reconcile state!")
-		glog.Warningf("Ignition passwd section contains changes")
-		return false, nil
+		// MCD will handle only ignition SSH keys
+		// This section will check (using some tbd magic) to see if the ONLY diff between 2
+		// Passwd configs is an SSHAuthorizedKey change
+		// Any change other than an SSHChanage will return false.
+
+		var onlySSHChanged bool
+		// If the size of list Users is different, a User has been added or deleted -> nope
+		if len(oldIgn.Passwd.Users) == len(newIgn.Passwd.Users) {
+			// Users is an array of User, iterate thru each
+			for i := range oldIgn.Passwd.Users {
+				// Check if the SSHKeys match
+				if !reflect.DeepEqual(oldIgn.Passwd.Users[i].SSHAuthorizedKeys, newIgn.Passwd.Users[i].SSHAuthorizedKeys) {
+					// keys don't match, let's create 2 temp Users
+					tempOld := oldIgn.Passwd.Users[i]
+					tempNew := newIgn.Passwd.Users[i]
+					// Let's set both Users' SSHKeys to match
+					tempOld.SSHAuthorizedKeys = tempNew.SSHAuthorizedKeys
+					// Now compare the two Users controlling for the SSHAuthorizedKey diff
+					if reflect.DeepEqual(tempOld, tempNew) {
+						// only diff between the two Passwds was the SSH key
+						onlySSHChanged = true
+						usersIndicesChanges = append(usersIndicesChanges, i)
+					} else {
+						// there are differences other than just the SSH key
+						// prob unnecessary since defaul bool value is false
+						onlySSHChanged = false
+						break
+					}
+				} else {
+					// SSH keys match, nothing to do on this
+					// prob unnecessary since defaul bool value is false
+					onlySSHChanged = false
+					break
+				}
+			}
+			// length of Users is different, we don't do those kinds of changes, so nope.
```

Comment:
I'm stuck between an attempted addition of a new user should noop (as it's doing here) or if we should raise the irreconcilable flag as the user was attempting to do something we are not going to do. @sdemos thoughts?